File: frontend/de/unika/ipd/grgen/ir/pattern/SubpatternUsage.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 5.0
 * Copyright (C) 2003-2020 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

/**
 * @author Edgar Jakumeit
 */

package de.unika.ipd.grgen.ir.pattern;

import java.util.List;

import de.unika.ipd.grgen.ir.Ident;
import de.unika.ipd.grgen.ir.Identifiable;
import de.unika.ipd.grgen.ir.Rule;
import de.unika.ipd.grgen.ir.expr.Expression;

public class SubpatternUsage extends Identifiable
{
	public Rule subpatternAction;
	public List<Expression> subpatternConnections;
	List<Expression> subpatternYields;

	public SubpatternUsage(String name, Ident ident, Rule subpatternAction,
			List<Expression> connections, List<Expression> yields)
	{
		super(name, ident);
		this.subpatternAction = subpatternAction;
		this.subpatternConnections = connections;
		this.subpatternYields = yields;
	}

	public Rule getSubpatternAction()
	{
		return subpatternAction;
	}

	public List<Expression> getSubpatternConnections()
	{
		return subpatternConnections;
	}

	public List<Expression> getSubpatternYields()
	{
		return subpatternYields;
	}
}

File: frontend/de/unika/ipd/grgen/ir/pattern/SubpatternUsage.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

/// <summary>
/// @author Edgar Jakumeit
/// </summary>

namespace de.unika.ipd.grgen.ir.pattern
{

using System.Collections.Generic;

using Ident = de.unika.ipd.grgen.ir.Ident;
using Identifiable = de.unika.ipd.grgen.ir.Identifiable;
using Rule = de.unika.ipd.grgen.ir.executable.Rule;
using Expression = de.unika.ipd.grgen.ir.expr.Expression;

public class SubpatternUsage : Identifiable
{
	public Rule subpatternAction;
	public IList<Expression> subpatternConnections;
	internal IList<Expression> subpatternYields;

	public SubpatternUsage(string name, Ident ident, Rule subpatternAction,
			IList<Expression> connections, IList<Expression> yields)
		: base(name, ident)
	{
		this.subpatternAction = subpatternAction;
		this.subpatternConnections = connections;
		this.subpatternYields = yields;
	}

	public virtual Rule SubpatternAction
	{
		get
		{
			return subpatternAction;
		}
	}

	public virtual IList<Expression> SubpatternConnections
	{
		get
		{
			return subpatternConnections;
		}
	}

	public virtual IList<Expression> SubpatternYields
	{
		get
		{
			return subpatternYields;
		}
	}
}

}
